### ISSUE TITLE
Application freezing

### ISSUE THREAD

**Issue by Despair2000** [softwarespartan/IB4m#64]
Hi Abel,

I have a problem I have a hard time to debug.

I have an application that makes extensive use of IB4m. It subscribes to 60+ stocks and performs all kind of operations. Everything works fine but sometimes (maybe every other day) the program freezes. If this happens I can't close it. Hitting CTRL-C will give me strange java errors I don't understand but doesn't stop the program. I really have to close matlab and restart it.
When I pause the running program at this point to come into matlabs debug-mode, I will usually end up in the processnotification.m and there I'm stuck. So when I try to trace forward from this point step by step or want to use the "step out" function to leave the process notification I can't. Looks little bit like calls to the process notification have accumulated on top of each other and leaving one just gives be back to another one.

I have no idea what is going on but it seems to be some part of IB4m that is causing this behavior. Do you have any idea?

**Comment by Despair2000** [comment]
I want to a for clarification, it is not only my application that freezes, it seems to be whole matlab. But let's wait and see how the days goes with 3GB heap space.

**Comment by Despair2000** [comment]
I watched it now the whole week and the problem seems to be solved. Increasing the java heap space to 3GB was the solution.
Still a little bit puzzled how my program can need so much memory but it works and this is the most important.
Closing the issue.

**Comment by softwarespartan** [comment]
As much as you are streaming, it is no surprise.  

Be sure to use the circular buffers where you can which will prevent too much accumulation when streaming.

Don’t hesitate to reopen issue if problem returns.

Cheers

**Comment by Despair2000** [comment]
> As much as you are streaming, it is no surprise. 
> 
> Be sure to use the circular buffers where you can which will prevent too much accumulation when streaming.

As much as I can see I only use circular buffers for the streaming. How can there be any accumulation? Shouldn’t the buffer simply throw out old observations when the specified capacity is reached and new data arrives?
And on top I’m clearing the buffers all the time (after I retrieved the data that has arrived).

Have a great weekend!

Boris

**Comment by softwarespartan** [comment]
@Despair2000 I pushed updates today that should help address this issue.  There were messages lingering too long in the internal event queue in the TWS message handler (under the hood in the jar file).  The processNotification.m was updated to directly delete the event once retrieved by the MATLAB event listener thread.  Note that I've not yet tested this on hardcore edge use-cases like multiple instances of MATLAB etc. so let me know if run into any issues.
